cfme/cloud/flavor.py

```diff
@@ -108,11 +145,75 @@ class Flavor(BaseEntity, Taggable):
     name = attr.ib()
     provider = attr.ib()
 
+    def delete(self, cancel=False):
+        """Delete current falvor """
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Remove Flavor', handle_alert=not cancel)
+        view.flash.assert_no_error()
+
+    @property
+    def vcpu(self):
```

Comment:
Bit of a miscommunication here I think.

So when creating a flavor in the collection, you pass these arguments: `ram`, `vcpus`, `disk`, `swap`, `rxtx`, `is_public`.

These are attributes of the flavor entity that you are modeling, and creating in the UI.

We don't want our model of the flavor entity to return the contents of the UI in its properties.
We want to use that model to CRUD the entity in MIQ, and then validate that the UI fields for the entity match the attributes of the model we tried to CRUD.

`ram`, `vcpus`, `disk`, `swap`, `rxtx`, `is_public` should all be class attributes, defined using `attr.ib()`, with defaults where necessary (is_public).

Then we can verify that `test_flavor.ram == view.entities.properties.get_text_of('Memory').split()[0]`